clash-prelude/src/Clash/Prelude.hs

```diff
@@ -130,8 +130,7 @@ module Clash.Prelude
   , Lift (..)
     -- ** Type classes
     -- *** Clash
-  -- , module Clash.Class.AutoReg
-  , autoReg, deriveAutoReg
+  , AutoReg, autoReg, deriveAutoReg
```

Comment:
Did you use tooling to discover this? It would be great if we could check export lists on CI - the issue seems to be coming up every few months.

Reply:
No, I ran into this when trying to use the implicit variant of `autoReg` in a polymorphic function and trying to write down its type.

If we could get the full export list programmatically, what would you want to check it with?

Reply:
Ah I figured you made this commit to get `Clash.Prelude` in line with `Clash.Explicit.Prelude`. What I'd like to do (at some point) is to keep a file encoding the differences between the Explicit Prelude and the Implicit Prelude. Then, if you'd add something to one Prelude but forget to add it to the other, CI could complain.

Reply:
But in that case; did you check whether the explicit prelude exports this too?

Reply:
Good idea

And yes, the explicit prelude [exports](https://github.com/clash-lang/clash-compiler/blob/e16e9012a19b0296bf0ace7c23ce44c200060e03/clash-prelude/src/Clash/Explicit/Prelude.hs#L110) the the full AutoReg module